homeassistant/components/sensor/nzbget.py

```diff
@@ -1,48 +1,66 @@
 
-Uses NZBGet's JSON-RPC API to query for monitored variables.
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.nzbget/
 """
 import logging
 from datetime import timedelta
+
 import requests
+import voluptuous as vol
 
+from homeassistant.const import (CONF_HOST, CONF_PASSWORD, CONF_USERNAME,
+                                 CONF_PLATFORM, CONF_NAME, CONF_PORT,
+                                 CONTENT_TYPE_JSON, CONF_MONITORED_VARIABLES)
 from homeassistant.helpers.entity import Entity
 from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+DEFAULT_NAME = 'NZBGet'
+DEFAULT_PORT = 6789
 
-REQUIREMENTS = []
 SENSOR_TYPES = {
-    "ArticleCacheMB": ("Article Cache", "MB"),
-    "AverageDownloadRate": ("Average Speed", "MB/s"),
-    "DownloadRate": ("Speed", "MB/s"),
-    "DownloadPaused": ("Download Paused", None),
-    "FreeDiskSpaceMB": ("Disk Free", "MB"),
-    "PostPaused": ("Post Processing Paused", None),
-    "RemainingSizeMB": ("Queue Size", "MB"),
+    'article_cache': ('Article Cache', 'MB'),
+    'average_download_rate': ('Average Speed', 'MB/s'),
+    'download_rate': ('Speed', 'MB/s'),
+    'download_paused': ('Download Paused', None),
+    'free_disk_space': ('Disk Free', 'MB'),
+    'post_paused': ('Post Processing Paused', None),
+    'remaining_size': ('Queue Size', 'MB'),
 }
+
 DEFAULT_TYPES = [
-    "DownloadRate",
-    "DownloadPaused",
-    "RemainingSizeMB",
+    'download_rate',
+    'download_paused',
+    'remaining_size',
 ]
 
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): 'nzbget',
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.string,
+    vol.Optional(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_MONITORED_VARIABLES, default=DEFAULT_TYPES):
+        [vol.In(SENSOR_TYPES.keys())],
+})
+
 _LOGGER = logging.getLogger(__name__)
 
 
-# pylint: disable=unused-argument
+# pylint: disable=unused-argument, too-many-locals
 def setup_platform(hass, config, add_devices, discovery_info=None):
-    """Set up nzbget sensors."""
-    base_url = config.get("base_url")
-    name = config.get("name", "NZBGet")
-    username = config.get("username")
-    password = config.get("password")
-    monitored_types = config.get("monitored_variables", DEFAULT_TYPES)
-
-    if not base_url:
-        _LOGGER.error("Missing base_url config for NzbGet")
-        return False
+    """Setup the NZBGet sensors."""
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    name = config.get(CONF_NAME)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    monitored_types = config.get(CONF_MONITORED_VARIABLES)
 
-    url = "{}/jsonrpc".format(base_url)
+    url = "http://{}:{}/jsonrpc".format(host, port)
```

Comment:
This is a breaking change that we should not forget to mention when this gets released.